sherpa/astro/ui/utils.py

```diff
@@ -10314,37 +10330,36 @@ def _get_stat_info(self):
     # Plotting
     ###########################################################################
 
-    def get_data_plot(self, id=None, recalc=True):
-        try:
-            d = self.get_data(id)
-        except IdentifierErr:
-            return super().get_data_plot(id, recalc=recalc)
+    def _get_plotobj(self, plottype, *, id=None, recalc=False):
+        """Select the plot object for the given plottype.
 
-        if isinstance(d, sherpa.astro.data.DataPHA):
-            plotobj = self._dataphaplot
+        The current mechanism to chose a specific type of object
+        is not particularly extensible.
+        """
+
+        # This is ugly as we repeat a lot of the logic of
+        # super()._get_plotobj.
+        #
```

Comment:
I've tried - in this PR and in previous attempts - a number of versions of this call and - as it is always unsatisfactory: we either have to repeat the same logic in most calls to this routine (e.g. the try/except block to access `data`), or we can include it here but there are some cases where it is not needed.

I have noted that this is not particularly extensible, but there is a benefit to a simple system (in earlier versions I used more-complicated classes, but they didn't add anything above the `if isinstance` chain of calls used below). One advantage to this simpler system is that it makes it more-obvious when we are adding extra logic (in particular the template-model/source plots).

Mentioned here, rather than below, is the related issue that when I over-ride the `get_foo_plot` call it can lead to the choice of either copying the logic from the superclass or deciding (for non PHA data) to call the superclass and re-run the call to `_get_plotobj`. Note that `get_fit_plot` and `get_model_component_plot`  show the two approaches.